algorithmic_efficiency/workloads/fastmri/fastmri_jax/workload.py

```diff
@@ -0,0 +1,288 @@
+    k1: float. Algorithm parameter K1 (see [1]).
+    k2: float. Algorithm parameter K2 (see [2]).
+
+  Returns:
+    mssim: float
+        The mean structural similarity index over the image.
+
+  References
+    [1] Wang, Z., Bovik, A. C., Sheikh, H. R., & Simoncelli, E. P.
+      (2004). Image quality assessment: From error visibility to
+      structural similarity. IEEE Transactions on Image Processing,
+      13, 600-612.
+      https://ece.uwaterloo.ca/~z70wang/publications/ssim.pdf,
+      :DOI:`10.1109/TIP.2003.819861`
+  """
+  filter_func = functools.partial(_uniform_filter, size=win_size)
+
+  num_points = win_size**len(im1.shape)
+
+  # filter has already normalized by num_points
+  cov_norm = num_points / (num_points - 1)  # sample covariance
+
+  # compute (weighted) means
+  ux = filter_func(im1)
+  uy = filter_func(im2)
+
+  # compute (weighted) variances and covariances
+  uxx = filter_func(im1 * im1)
+  uyy = filter_func(im2 * im2)
+  uxy = filter_func(im1 * im2)
+  vx = cov_norm * (uxx - ux * ux)
+  vy = cov_norm * (uyy - uy * uy)
+  vxy = cov_norm * (uxy - ux * uy)
+
+  c1 = (k1 * data_range)**2
+  c2 = (k2 * data_range)**2
+
+  a1 = 2 * ux * uy + c1
+  a2 = 2 * vxy + c2
+  b1 = ux**2 + uy**2 + c1
+  b2 = vx + vy + c2
+
+  d = b1 * b2
+  s = (a1 * a2) / d
+
+  # to avoid edge effects will ignore filter radius strip around edges
+  pad = (win_size - 1) // 2
+
+  # compute (weighted) mean of ssim.
+  return jnp.mean(s.at[pad:-pad, pad:-pad].get())
+
+
+def ssim(logits, targets, mean=None, std=None, volume_max=None):
+  """Computes example-wise structural similarity for a batch.
+
+  NOTE(dsuo): we use the same (default) arguments to `structural_similarity`
+  as in https://arxiv.org/abs/1811.08839.
+
+  Args:
+   logits: (batch,) + input.shape float array.
+   targets: (batch,) + input.shape float array.
+   mean: (batch,) mean of original images.
+   std: (batch,) std of original images.
+   volume_max: (batch,) of the volume max for the volumes each example came
+    from.
+  Returns:
+    Structural similarity computed per example, shape [batch, ...].
+  """
+  if volume_max is None:
+    volume_max = jnp.ones(logits.shape[0])
+
+  # NOTE(dsuo): `volume_max` can be 0 if we have a padded batch, but this will
+  # lead to NaN values in `ssim`.
+  volume_max = jnp.where(volume_max == 0, jnp.ones_like(volume_max), volume_max)
+
+  if mean is None:
+    mean = jnp.zeros(logits.shape[0])
+
+  if std is None:
+    std = jnp.ones(logits.shape[0])
+
+  mean = mean.reshape((-1,) + (1,) * (len(logits.shape) - 1))
+  std = std.reshape((-1,) + (1,) * (len(logits.shape) - 1))
+  logits = logits * std + mean
+  targets = targets * std + mean
+  print(logits.shape, targets.shape, volume_max.shape)
+  ssims = jax.vmap(structural_similarity)(logits, targets, volume_max)
+  return ssims
+
+
+class FastMRIWorkload(BaseFastMRIWorkload):
+
+  def __init__(self):
+    super().__init__()
+    self._param_types = None
+    self._eval_iters = {}
```

Comment:
`_param_types` and `_eval_iters` can be moved to `BaseFastMRIWorkload`, since they are also used in the PyTorch version of the workload.